script/generate-release-notes.ts

```diff
@@ -1,226 +1,210 @@
+import * as glob from 'glob'
+import { basename } from 'path'
+import * as fs from 'fs'
 import { Octokit } from '@octokit/rest'
-import fetch from 'node-fetch'
-
-// five targeted OS/arch combinations
-// two files for each targeted OS/arch
-// two checksum files for the previous
-const SUCCESSFUL_RELEASE_FILE_COUNT = 4 * 2 * 2
-
-process.on('unhandledRejection', reason => {
-  console.log(reason)
-})
-
-async function getBuildUrl(
-  octokit: Octokit,
-  owner: string,
-  repo: string,
-  ref: string
-) {
-  const response = await octokit.repos.listCommitStatusesForRef({
-    owner,
-    repo,
-    ref,
-  })
-
-  // Travis kicks off this build after a tag is pushed to the repository
-  const statuses = response.data
-  const travisStatus = statuses.find(
-    s => s.context === 'continuous-integration/travis-ci/push'
-  )
-
-  if (travisStatus == null) {
-    const contexts = statuses.map(s => s.context)
-    console.log(
-      `👀 Uh-oh, I couldn't find the right commit status. Found these contexts: ${JSON.stringify(
-        contexts
-      )}`
-    )
-    console.log(
-      `Please open an issue against https://github.com/desktop/dugite-native so it can be fixed!`
-    )
-  } else {
-    console.log(
-      `👀 Follow along with the build here: ${travisStatus.target_url}`
-    )
-  }
-}
 
-async function run() {
-  const token = process.env.GITHUB_ACCESS_TOKEN
-  if (token == null) {
-    console.log(`🔴 No GITHUB_ACCESS_TOKEN environment variable set`)
-    return
-  }
+export default class GenerateReleaseNotes {
+  // five targeted OS/arch combinations
+  // two files for each targeted OS/arch
+  // two checksum files for the previous
+  private SUCCESSFUL_RELEASE_FILE_COUNT = 4 * 2 * 2
+  private args = process.argv.slice(2)
+  private expectedArgs = [
+    {
+      key: 0,
+      name: 'artifactsDir',
+      description: 'full path to the artifacts directory',
+    },
+    {
+      key: 1,
+      name: 'tagName',
+      description:
+        'name of the GitHub tag that we use to generate the changelog',
+    },
+    {
+      key: 2,
+      name: 'githubToken',
+      description: 'GitHub API token',
+    },
+  ]
+  private expectedArgsString = this.expectedArgs
+    .map(arg => `\${${arg.name}}`)
+    .join(' ')
+
+  /**
+   * Full path to the artifacts directory
+   */
+  private artifactsDir: string
+
+  /**
+   * Name of the GitHub tag that we use to generate the changelog
+   */
+  private tagName: string
+
+  /**
+   * GitHub API token
+   */
+  private githubToken: string
+
+  private owner = 'dennisameling'
```

Comment:
Note to self: change this to `desktop` (and in the future can be received from CI as well)